tests/testthat/test-library_call_linter.R

```diff
@@ -233,50 +233,51 @@ test_that("skips allowed usages of library()/character.only=TRUE", {
 
 test_that("blocks disallowed usages of strings in library()/require()", {
   linter <- library_call_linter()
+  char_only_msg <- rex::rex("Use symbols in library calls", anything, "character.only")
+  direct_stub <- "Call library() directly, not vectorized with "
 
   expect_lint(
     'library("data.table")',
     rex::rex("Use symbols, not strings, in library calls."),
     linter
   )
 
-  expect_lint(
-    'library("data.table", character.only = TRUE)',
-    rex::rex("Use symbols in library calls", anything, "character.only"),
-    linter
-  )
-
-  expect_lint(
-    'suppressWarnings(library("data.table", character.only = TRUE))',
-    rex::rex("Use symbols in library calls", anything, "character.only"),
-    linter
-  )
+  expect_lint('library("data.table", character.only = TRUE)', char_only_msg, linter)
+  expect_lint('suppressWarnings(library("data.table", character.only = TRUE))', char_only_msg, linter)
 
   expect_lint(
     "do.call(library, list(data.table))",
-    rex::rex("Call library() directly, not vectorized with do.call()"),
+    rex::rex(direct_stub, "do.call()"),
     linter
   )
-
   expect_lint(
     'do.call("library", list(data.table))',
-    rex::rex("Call library() directly, not vectorized with do.call()"),
+    rex::rex(direct_stub, "do.call()"),
     linter
   )
-
   expect_lint(
     'lapply("data.table", library, character.only = TRUE)',
-    rex::rex("Call library() directly, not vectorized with lapply()"),
+    rex::rex(direct_stub, "lapply()"),
     linter
   )
-
   expect_lint(
     'purr::map("data.table", library, character.only = TRUE)',
-    rex::rex("Call library() directly, not vectorized with map()"),
+    rex::rex(direct_stub, "map()"),
     linter
   )
 })
 
+test_that("character.only=TRUE is caught in *apply functions passed as strings", {
```

Comment:
Is it worth switching to parametrized tests here?

Reply:
Hmm my sense is it's a lot less readable:

```r
local({
  linter <- library_call_linter()

  cases <- expand.grid(
    call = c("library", "require"),
    quote = c("'", '"'),
    stringsAsFactors = FALSE
  )

  cases$lint_msg <- vapply(paste("Call", cases$call, "directly, not vectorized with sapply()"), rex::rex, character(1L))
  cases$.test_name <- sprintf("Attach with %s, quote with %s", cases$call, cases$quote)

  patrick::with_parameters_test_that(
    "character.only=TRUE is caught in *apply functions passed as strings",
    {
      expect_lint(
        sprintf("sapply(pkgs, %s, character.only = TRUE)", sQuote(call, quote)), 
        lint_msg,
        linter
      )
    },
    .cases = cases
  )
})
```